bin/compile

```diff
@@ -452,8 +452,16 @@ case "${TOOL}" in
         unset GIT_DIR # unset git dir or it will mess with goinstall
         cd ${build}
         export GOBIN="${build}/bin"
+        if [[ -x bin/pre-compile ]]; then
+            echo "-----> Running bin/pre-compile hook"
+            sub_env bin/pre-compile
```

Comment:
I've been struggling with the usefulness of providing `sub_env` hooks to users on the python buildpack and have been considering deprecating/finding another way to serve these needs, particularly because it results in surprising behavior like environment vars not being shared with the parent process. It also might be of note that for generating static assets, users are not using hooks, but functionality is present to explicitly generate the static assets (although if I had to do it over, I'd probably direct users to use release phase for that)

TLDR I don't think I have enough context on what tools or why you'd want to optimize your slug size by not installing tools (are these big tools?) but still want to use them, and I'm not 100% certain that a `sub_env` process successfully persists the changes those tools would make

Reply:
> (although if I had to do it over, I'd probably direct users to use release phase for that)

Release phase cannot modify the slug, and should not be used for generating static assets. It can be used to upload static assets to a CDN, but the compilation should happen at build time.

> it results in surprising behavior like environment vars not being shared with the parent process

Ex: If I understand correctly, a script installing a binary can't modify the `PATH` env var in these hooks when they are called with `sub_env` and this is surprising to users

Reply:
Okay, then why does the python buildpack use it and why is it part of the buildpack stdlib?

Reply:
That was added by a previous maintainer and is only used by the python buildpack afaict. My understanding is that we will likely remove it since it's caused issues and confusion for users.